### DISCUSSION TITLE
PostgreSQL (and Minio) K8s operators

### DISCUSSION THREAD

**Discussion by josefhandl** [neicnordic/sensitive-data-archive#840] [Ideas]
Hello,

I would like to ask, or come up with an idea to use [CloudNativePG](https://cloudnative-pg.io/documentation/1.23/) operator for Kubernetes to manage Postgres instead of having custom Postgres Pod. It could save the work with implementing and maintaining a custom Postgres solution dedicated to this project.

We use it for several applications, and I have experimented with it also for the SDA in our fork (https://github.com/CERIT-SC/sensitive-data-archive/tree/cerit-prototype/charts/sda-db-cnpg). The early prototype of our experiment it works fine, but of course I am not able to consider the requirements, security aspects, and future needs of this project.

The same case could be to use [Minio operator](https://min.io/docs/minio/kubernetes/upstream/operations/install-deploy-manage/deploy-minio-tenant.html) as a S3 storage.

**Comment by jbygdell** [comment]
Hi I have started looking at the CloudNativePg for our production setup, there are a few issues still to be ironed out. As for the postgres container found in the repo, for my part it is used mostly for development right now.

For in cluster storage we use [rook](https://rook.io/) as it handles all possible use cases (filesystem, block, object), archive storage on the other hand are external due to the expected amount of data.
